src/system_tables.rs

```diff
@@ -282,3 +309,85 @@ impl SeafowlSystemTable for TablePartitionsTable {
             .map_err(DataFusionError::from)
     }
 }
+
+// Table listing all dropped tables that are pending lazy deletion on subsequent `VACUUM`s
+struct DroppedTablesTable {
+    database: Arc<str>,
+    schema: SchemaRef,
+    table_catalog: Arc<dyn TableCatalog>,
+}
+
+impl DroppedTablesTable {
+    fn new(database: Arc<str>, table_catalog: Arc<dyn TableCatalog>) -> Self {
+        Self {
+            // This is dictated by the output of `get_dropped_tables`, except that we omit the
+            // database_name field, since we scope down to the database at hand.
+            database,
+            schema: Arc::new(Schema::new(vec![
+                Field::new("table_schema", DataType::Utf8, false),
+                Field::new("table_name", DataType::Utf8, false),
+                Field::new("uuid", DataType::Utf8, false),
+                Field::new("deletion_status", DataType::Utf8, false),
+                Field::new(
+                    "drop_time",
+                    // TODO: should we be using a concrete timezone here?
```

Comment:
🤷 could split the difference and return the server's local timezone (if SQLite even supports storing timestamps like that) or following what the rest of Delta does

Reply:
Oh this is just for our representation of the system table; the underlying catalog column defaults to `strftime('%s','now')` (for SQLite) and uses UTC.

I'll remove the comment as it's not really that important.